dpctl/tensor/libtensor/include/kernels/elementwise_functions/exp.hpp

```diff
@@ -64,7 +64,58 @@ template <typename argT, typename resT> struct ExpFunctor
 
     resT operator()(const argT &in)
     {
-        return std::exp(in);
+        if constexpr (is_complex<argT>::value) {
+            using realT = typename argT::value_type;
+
+            constexpr realT q_nan = std::numeric_limits<realT>::quiet_NaN();
+
+            const realT x = std::real(in);
+            const realT y = std::imag(in);
+            if (std::isfinite(x)) {
+                if (std::isfinite(y)) {
+                    return std::exp(in);
+                }
+                else {
+                    return resT{q_nan, q_nan};
+                }
+            }
+            else if (std::isnan(x)) {
+                /* x is nan */
+                if (y == realT(0)) {
+                    return resT{in};
+                }
+                else {
+                    return resT{x, q_nan};
+                }
+            }
+            else {
+                if (x > realT(0)) { /* x is +inf */
+                    if (y == realT(0)) {
+                        return resT{x, y};
+                    }
```

Comment:
There is not point in checking for zero here, it will be handled by the `if (std::isfinite(y))` clause.

Reply:
I think it is needed since `x * std::sin(y)` in `if (std::isfinite(y))` clause will be infinity multiplied by zero for this case which is undefined, so we need to address it separately.

Reply:
Yes, alright. That branch is needed.